debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -122,30 +124,43 @@ public PostgresConnection executeWithoutCommitting(String... statements) throws
         return ServerInfo.ReplicaIdentity.parseFromDB(replIdentity.toString());
     }
 
-    protected ServerInfo.ReplicationSlot readReplicationSlotInfo(String slotName, String pluginName) throws SQLException {
-        AtomicReference<ServerInfo.ReplicationSlot> replicationSlotInfo = new AtomicReference<>();
-        String database = database();
-        prepareQuery(
-             "select * from pg_replication_slots where slot_name = ? and database = ? and plugin = ?", statement -> {
-                 statement.setString(1, slotName);
-                 statement.setString(2, database);
-                 statement.setString(3, pluginName);
-             },
-             rs -> {
-                 if (rs.next()) {
-                     boolean active = rs.getBoolean("active");
-                     Long confirmedFlushedLSN = parseConfirmedFlushLsn(slotName, pluginName, database, rs);
-                     replicationSlotInfo.compareAndSet(null, new ServerInfo.ReplicationSlot(active, confirmedFlushedLSN));
-                 }
-                 else {
-                     LOGGER.debug("No replication slot '{}' is present for plugin '{}' and database '{}'", slotName,
-                                  pluginName, database);
-                     replicationSlotInfo.compareAndSet(null, ServerInfo.ReplicationSlot.INVALID);
-                 }
-             }
-        );
+    protected ServerInfo.ReplicationSlot readReplicationSlotInfo(String slotName, String pluginName) throws SQLException, InterruptedException {
+        final int maxAttempts = 30;
```

Comment:
I'll make these constants for the sake of comprehensibility. Regarding @jchipmunk's suggestion of making them configurable, I'd say let's wait whether there's demand. In general I prefer to expose as less knobs and options of this kind as possible, in order to keep things understandable for users. Having a gazillion of options can be intimidating :)

That said we might increase the hard-coded period here. How about extending it to half an hour (keeping the 2 sec interval)? Then we should be on the safe side, barring any long running TX such as batch jobs. But in such case users can be advised to workaround by not starting the connector which such long running process is active. Any thoughts?